src/plugins/openmeteo/datasource.py

```
@@ -0,0 +1,333 @@
+                url = f"{self.base_url}/archive"
+                
+                async with session.get(url, params=params) as response:
+                    if response.status != 200:
+                        logger.error(f"Open-Meteo API error: {response.status}")
+                        current_start = chunk_end + timedelta(days=1)
+                        continue
+                        
+                    data = await response.json()
+                    
+                if 'hourly' not in data:
+                    logger.warning(f"No hourly data in response for {sensor.location.name}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)